facilitator/src/transport/gcs.rs

```diff
@@ -419,6 +430,13 @@ impl StreamingTransferWriter {
                     ))?
                     .parse::<usize>()
                     .context("End in range header {} not a valid usize")?;
+                // end is usize and so parse would fail if the value in the
+                // header was negative, but we still defend ourselves against
+                // it being bigger than is possible given our position in the
+                // object.
+                if end >= object_length_thus_far {
```

Comment:
Now that we've switched up uploading 8MB at a time, I think this check should be against the amount that we actually uploaded - I.e. `self.object_upload_position + body.len() - 1` from above (which can be extracted into a variable). I think `object_length_thus_far` becomes unneeded.

Also, looking at this again, I think we can check for `end > object_upload_position`, i.e. the end of the range should never get smaller. Arguably that could be a `>=`, but if it's `>` that means we made no progress with this chunk, which I think would be some sort of error rather than a 308.